task_netcore/task4/WebApi/task4/task4.BLL/Services/RatingService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using AutoMapper;
+using System.Collections.Generic;
+using System.Linq;
+using task4.BLL.Interfaces;
+using task4.BLL.Models;
+using task4.DAL.Entities;
+using task4.DAL.Interfaces;
+
+namespace task4.BLL.Services
+{
+    public class RatingService: IRatingService
+    {
+        private readonly IUnitOfWork dataBase;
+        private readonly IMapper _mapper;
+        private readonly IMovieService _movieService;
+
+        public RatingService(IUnitOfWork uow, IMapper mapper, IMovieService movieService)
+        {
+            dataBase = uow;
+            _mapper = mapper;
+            _movieService = movieService;
+        }
+
+        public RatingResultModel AddRating(RatingModel ratingModel)
+        {
+            var userRating = dataBase.RatingRepository.GetQueryableAll().FirstOrDefault(r => r.User.Id == ratingModel.UserId && r.Movie.Id == ratingModel.MovieId);
+            var ratingResultModel = new RatingResultModel();
+
+            if (userRating != null)
+            {
+                ratingResultModel.Errors = new List<string>();
+                ratingResultModel.Errors.Add("You can't rate again");
```

Comment:
Could merge declaration and initialization `ratingResultModel.Errors = new List<string> {"You can't rate again"};`